compiler/suggest.nim

```diff
@@ -251,11 +251,18 @@ proc `$`*(suggest: Suggest): string =
       result.add($suggest.endCol)
 
 proc suggestToSuggestInlayHint*(sug: Suggest): SuggestInlayHint =
+
+  func convertTypeName2InlayHint(s: string): string =
+    if s.startsWith("int literal("):
+      "int"
+    else:
+      s
+
```

Comment:
That's a hack. Instead tell `typeToString` to ignore the literal or something like that.

Reply:
Good idea. I introduced a new value preferInlayHint and removed the hack.